src/c2goto/library/mman.c

```diff
@@ -0,0 +1,33 @@
+#include <stdlib.h>
+#include <sys/mman.h>
+
+#undef mmap
+#undef munmap
+
+void *mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset)
+{
+__ESBMC_HIDE:;
+  (void *)addr;
+  (void)prot;
+  (void)flags;
+  (void)fd;
+  (void)offset;
+  void *res = malloc(length);
```

Comment:
You should check for the [ERRORS](https://man7.org/linux/man-pages/man2/mmap.2.html#ERRORS)

For instance, there are errors for when the `length` is 0 and when the `flags` is not valid

Reply:
Are these casts (e.g., (void)prot;) really needed?

Reply:
These casts are there to avoid the unused parameter warning.

I agree with Rafael, we should properly model these functions.